File: src/1403-Minimum-Subsequence-in-Non-Increasing-Order/1403.java
class Solution {
    public List<Integer> minSubsequence(int[] nums) {
        Arrays.sort(nums);
        List<Integer> res = new ArrayList();
        int cur = 0, t = 0;
        
        for (int i : nums) t += i;
        t /= 2;
        
        for (int i = nums.length - 1; i >= 0; i--) {
            cur += nums[i];
            res.add(nums[i]);
            if (cur > t) return res;
        }
        return res;
    }
}

File: src/1403-Minimum-Subsequence-in-Non-Increasing-Order/1403.js
var minSubsequence = function(nums) {
    nums.sort((a, b) => a - b);
    let res = [], cur = 0, t = 0;
    
    for (let i of nums) t += i;
    t >>= 1;
    
    for (let i = nums.length - 1; i >= 0; i--) {
        cur += nums[i];
        res.push(nums[i]);
        if (cur > t) return res;
    }
    return res;
};